applications/cam360/cam360.js

```diff
@@ -0,0 +1,605 @@
+            var noRoll = Quat.cancelOutRoll(tabletRotation); //Pushing the roll is getting quite complexe
+            euler = Quat.safeEulerAngles(noRoll);
+            yaw = euler.y - 180;
+            if (yaw < -180) { yaw = yaw + 360;}
+            yaw = -yaw;
+            pitch = euler.x;
+            roll = 0;
+        }
+
+        tablet.emitScriptEvent(JSON.stringify({
+            "channel": channel,
+            "method": "yawPitchRoll",
+            "yaw": yaw,
+            "pitch": pitch,
+            "roll": roll
+        })); 
+        
+    }
+
+    function cleanup() {
+
+        if (appStatus) {
+            tablet.gotoHomeScreen();
+            tablet.webEventReceived.disconnect(onAppWebEventReceived);
+            if (!udateSignateDisconnected) {
+                Script.update.disconnect(myTimer);
+                udateSignateDisconnected = true;
+            }
+        }
+
+        tablet.screenChanged.disconnect(onScreenChanged);
+        tablet.removeButton(button);
+        
+        rpo360Off();
+        
+        if (takePhotoControllerMapping) {
+            takePhotoControllerMapping.disable();
+        }
+        
+        Window.domainChanged.disconnect(onDomainChanged);
+        Window.snapshot360Taken.disconnect(on360SnapshotTaken);
+        HMD.displayModeChanged.disconnect(onHMDChanged);
+    }
+
+    Script.scriptEnding.connect(cleanup);
+ 
+    //controller 
+    function setTakePhotoControllerMappingStatus() {
+        if (!takePhotoControllerMapping) {
+            return;
+        }
+        if (!isThrowMode) {
+            takePhotoControllerMapping.enable();
+        } else {
+            takePhotoControllerMapping.disable();
+        }
+    }
+
+    var takePhotoControllerMapping;
+    var takePhotoControllerMappingName = 'Overte-cam360-Mapping-Capture';
+    function registerTakePhotoControllerMapping() {
+        takePhotoControllerMapping = Controller.newMapping(takePhotoControllerMappingName);
+        if (controllerType === "OculusTouch") {
+            takePhotoControllerMapping.from(Controller.Standard.RS).to(function (value) {
+                if (value === 1.0) {
+                    if (camera) {
+                        capture();
+                    }
+                }
+                return;
+            });
+        } else if (controllerType === "Vive") {
+            takePhotoControllerMapping.from(Controller.Standard.RightPrimaryThumb).to(function (value) {
+                if (value === 1.0) {
+                    if (camera) {
+                        capture();
+                    }
+                }
+                return;
+            });
+        }
+    }
+
+    var controllerType = "Other";
+    function registerButtonMappings() {
+        var VRDevices = Controller.getDeviceNames().toString();
+        if (VRDevices) {
+            if (VRDevices.indexOf("Vive") !== -1) {
+                controllerType = "Vive";
```

Comment:
We will need to update this so it works with other devices too, but we can do it in the separate commit.

Reply:
Just for reference, "Vive" probably means any OpenVR device in practice, because we still use legacy bindings (from the time when the Vive was the only OpenVR device) that SteamVR provides backwards compatibility for.

Reply:
I was doing the same lecture of this. If this becomes a problem for any new devices we will have to re-adjust this (with probably a lot of other scripts too)